m3u2xspf.py

```diff
@@ -29,16 +32,33 @@ def get_tags(path):
     else:
         return {'location': path}
 
+import re
 
 def iter_lines(path):
     root = os.path.dirname(path)
+    track= {}
+    target = 'M3U'
     with open(sys.argv[1]) as fh:
         for line in fh:
             line = line.rstrip()
-            if line.startswith('#'):
+            if line.startswith('#EXTM3U'):
+                 target = 'M3UEXT'   
+            elif line.startswith('#EXTINF'):
+                matches = re.compile('#EXTINF:(\d*),([^,]+),(?:logo=(.*))?', re.IGNORECASE).findall(line) 
```

Comment:
This line carries most of the weight, so it deserves most scrutiny. Unfortunately there is no real spec for M3U and M3U spec (the closest thing we have is [rfc8216](https://www.rfc-editor.org/rfc/rfc8216)), so there is no "correct" way to do this. Still I think we can improve it in some ways:

- the logo property is not mentioned in rfc8216. I have mostly seen something like this in IPTV files, but in a different format. Where did you find this format? I am not opposed to including it, but it should be fully optional ,i.e. the `,` needs to be inside the optional block.
- rfc8216 says that this should be case sensitive and I have never seen `#extinf` in the wild, so I think we can remove the `re.IGNORECASE`.
- It is better to use `fullmatch()` here instead of `findall()`. 
- If you want to compile the regular expression it should be done once outside of the loop, otherwise you can just do `re.fullmatch(pattern, line)`.
- If the line does not match the pattern it should be ignored.